merkle-tree/merkle_tree_benchmark_test.go

```diff
@@ -0,0 +1,67 @@
+package merkle
+
+import (
+	"testing"
+
+	"github.com/0xPolygon/polygon-edge/helper/common"
+	"github.com/stretchr/testify/require"
+)
+
+func Benchmark_MerkleTreeCreation(b *testing.B) {
+	const numOfLeaves = 10_000
```

Comment:
You could have created different test cases and variate number of leaves. That way we could have better understanding how algorithm performs, since we would have more results to compare against and therefore we could conclude what is are the time and space complexity.

You can take a look to this as a reference:
https://github.com/maticnetwork/bor/pull/262/files#diff-a424e26a1e19cffe181aa20565feb4f3dc617fc9e6e54192351555517f5c92f0R36-R73

Reply:
Isn't that fuzz testing?

Reply:
No, it isn't. I'm not saying that you should randomize tree size, but just have more than single test case so that you are able to see how algorithm behaves for different data sets.

Reply:
Ok. I'll add more cases for different tree sizes.